extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -901,7 +902,14 @@ private static void checkPlayabilityStatus(@Nonnull final JsonObject playability
             }
         }
 
-        throw new ContentNotAvailableException("Got error: \"" + reason + "\"");
+        // "Sign in to confirm that you're not a bot"
+        if (reason != null && reason.contains("a bot")) {
+            throw new SignInConfirmNotBotException(
+                    "YouTube probably temporarily blocked this IP, got error "
+                            + status + ": \"" + reason + "\"");
+        }
+
+        throw new ContentNotAvailableException("Got error " + status + ": \"" + reason + "\"");
```

Comment:
I was thinking, should this really be a `ContentNotAvailableException`? I think `ContentNotAvailableException` should be reserved for when the extractor notices that the service explicitly reports a resource as not being available (e.g. age restricted, channel deleted, and so on). It should not be used for unknown errors in fetching resources, which is the case here. I would turn this into just `ParsingException`, add a clarifying comment to `ContentNotAvailableException`, and check all other places where `ContentNotAvailableException` is thrown. What's your opinion @AudricV?

Note: currently in some parts of NewPipe we assume that `ContentNotAvailableException`s are not supposed to be reported.

Reply:
Unless we start detecting all real unavailability errors (deleted video, unavailable/removed due to closed/removed YouTube account, ...), I don't think so.